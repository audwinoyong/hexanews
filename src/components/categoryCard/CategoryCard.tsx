import * as React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import { History } from 'history'

import placeholder from '../../assets/placeholder.png'
import styles from './styles'

type Props = {
  category: string,
  history: History,
}

const CategoryCard: React.StatelessComponent<Props> = ({
  category,
  history
}) => {
  return (
    <div style={styles.root}>
      <Card style={styles.card} onClick={() => history.push(`/categories/${category}`)}>
        <CardMedia
          style={styles.media}
          image={placeholder}
        />
        <CardContent>
          <Typography variant='title'>
            {category.charAt(0).toUpperCase() + category.slice(1)}
          </Typography>
        </CardContent>
      </Card>
    </div>
  )
}

export default CategoryCard