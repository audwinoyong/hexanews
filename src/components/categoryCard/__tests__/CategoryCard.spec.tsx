import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

import CategoryCard from '../CategoryCard'

Enzyme.configure({ adapter: new Adapter() })

describe('Category Card', () => {
  it('renders Category Card', () => {
    const component = renderer.create(
      <CategoryCard
        category='technology'
        dispatch={jest.fn()}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handle button go to category detail', () => {
    const dispatch = jest.fn()
    const component = (
      <CategoryCard
        category='technology'
        dispatch={dispatch}
      />
    )
    const layout = shallow(component)
    layout.find('#category-technology').first().simulate('click')
    expect(dispatch).toHaveBeenCalled()
  })
})