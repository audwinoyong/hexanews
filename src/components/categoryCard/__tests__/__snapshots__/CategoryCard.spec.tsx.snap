// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Card renders Category Card 1`] = `
<div
  style={
    Object {
      "cursor": "pointer",
      "flexGrow": 1,
    }
  }
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
    id="category-technology"
    onClick={[Function]}
    style={
      Object {
        "boxShadow": "0px 0.25px 0.35px 0.35px rgba(0, 0, 0, 0.1)",
        "maxWidth": 300,
      }
    }
  >
    <div
      className="MuiCardMedia-root-29"
      src={undefined}
      style={
        Object {
          "height": 0,
          "paddingTop": "56.25%",
        }
      }
    />
    <div
      className="MuiCardContent-root-31"
    >
      <h2
        className="MuiTypography-root-32 MuiTypography-title-38"
      >
        Technology
      </h2>
    </div>
  </div>
</div>
`;
